lib/archethic/mining/transaction_context.ex

```diff
@@ -98,23 +91,26 @@ defmodule Archethic.Mining.TransactionContext do
   end
 
   defp request_previous_tx(previous_address, nodes) do
-    Task.Supervisor.async(
-      TaskSupervisor,
-      fn ->
-        # Timeout of 4 sec because the coordinator node wait 5 sec to get the context
-        # from the cross validation nodes
-        case TransactionChain.fetch_transaction(previous_address, nodes,
-               search_mode: :remote,
-               timeout: 4000
-             ) do
-          {:ok, tx} ->
-            tx
-
-          {:error, _} ->
-            nil
-        end
+    Task.Supervisor.async(TaskSupervisor, fn ->
+      # Timeout of 2.5 sec because the coordinator node wait 3 sec to get the context
+      # from the cross validation nodes
+      case TransactionChain.fetch_transaction(previous_address, nodes,
+             search_mode: :remote,
+             timeout: 2500
```

Comment:
I'm wondering if it should be based on the variable of the coordinator waiting time to be more dynamic.

Reply:
Deleted the commit, we'll need to rework globally on mining timeout